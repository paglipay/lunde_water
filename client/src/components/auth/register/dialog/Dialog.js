import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Button, Header, Modal } from 'semantic-ui-react';

const Dialog = () => {
  const [open, setOpen] = useState(false);

  return (
    <Modal
      onClose={() => setOpen(false)}
      onOpen={() => setOpen(true)}
      open={open}
      trigger={<Button>Sign Up</Button>}
    >
      <Modal.Header>Sign Up</Modal.Header>
      <Modal.Content>
        <Modal.Description>
          <Header>Register with Lunde Water</Header>
          <p>Form</p>
        </Modal.Description>
      </Modal.Content>
      <Modal.Actions>
        <Button color='black' onClick={() => setOpen(false)}>
          Close
        </Button>
      </Modal.Actions>
    </Modal>
  );
};

export default Dialog;
