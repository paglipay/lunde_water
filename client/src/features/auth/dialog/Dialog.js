import React, { useState } from 'react';
import Login from '../login/form/Login';
import { Button, Header, Modal } from 'semantic-ui-react';

const Dialog = () => {
  const [open, setOpen] = useState(false);

  return (
    <Modal
      onClose={() => setOpen(false)}
      onOpen={() => setOpen(true)}
      open={open}
      trigger={<Button>Login</Button>}
    >
      <Modal.Header>Lunde Water</Modal.Header>
      <Modal.Content>
        <Modal.Description>
          <Header>Login to your account</Header>
          <Login/>
        </Modal.Description>
      </Modal.Content>
      <Modal.Actions>
        <Button color='black' onClick={() => setOpen(false)}>
          Close
        </Button>
      </Modal.Actions>
    </Modal>
  );
};

export default Dialog;
