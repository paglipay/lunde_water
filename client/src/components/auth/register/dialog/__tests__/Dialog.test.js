import React from 'react';
import { render } from '@testing-library/react';
import Dialog from '../Dialog';
import renderer from 'react-test-renderer';
import Portal from 'semantic-ui-react/dist/commonjs/addons/Portal/Portal';
import { Modal } from 'semantic-ui-react';


describe('Register dialog component', () => {
  it('should render without crashing', () => {
    render(<Dialog />);
  });

  it('should display a sign up button', () => {
    const { getByText } = render(<Dialog />);
    const button = getByText('Sign Up');

    expect(button).toBeTruthy();
  });
});

jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children);

describe('<Modal />', () => {
  test('snapshot', () => {
    const tree = renderer.create(<Modal open>Inside my modal!</Modal>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
