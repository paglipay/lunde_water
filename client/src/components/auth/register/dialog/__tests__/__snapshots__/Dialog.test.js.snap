// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> snapshot 1`] = `
<div
  className="ui page modals dimmer transition visible active"
>
  <div
    className="ui modal transition visible active"
    style={Object {}}
  >
    Inside my modal!
  </div>
</div>
`;
